### DISCUSSION TITLE
Where do Control Rig variables get modified?

### DISCUSSION THREAD

**Discussion by DBTech77** [Sixze/ALS-Refactored#158] [Q&A]
I've been working on retargeting this to the UE4 Mannequin because a lot of my characters and other animations are based on that skeleton, I have pretty much everything working except the control rig portion for the feet/hand ik

Whenever the control rig node is hooked up in the main animation graph my characters legs move really weird and funny, Im wondering if it has something to do with these variables?

![Untitled](https://user-images.githubusercontent.com/83677055/192651743-76746623-455a-4f07-b253-e061bf1e4967.png)

In the video below you can see the feet placement seems to work properly but the leg movement is all wonky

https://user-images.githubusercontent.com/83677055/192653909-77893b23-ec3e-486d-9763-06213f016553.mp4

I've copied all logic and variables from the CR_Als rig to a new control rig based on the ue4 mannequin with the proper bones and switched the node in the main anim graph to look to my new rig but it doesn't make a difference. Any help is appreciated!

**Comment by DBTech77** [comment]
I still havent figured out how to get the control rig to work right but for anyone else with this issue, I found this simple foot ik that works great with everything else! https://www.unrealengine.com/marketplace/en-US/product/basic-foot-ik-system

**Comment by nekwo** [comment]
If you have a skeletal mesh with the same epic bones (same names and bone length proportions) and  you re-assigned the mesh to the als skeleton then you can just use the same control rig seamlessly otherwise you will have to redo the control rig to the new bone proportions (and names if different) then change the control rig reference in the animbp.

**Comment by DefSteam** [comment]
I had the same problem today with a custom mesh using the Epic skeleton. Turned out to be the hidden retargeting settings. Changing everything from the pelvis down to 'Orient and Scale' and then 3 foot IK bones to 'Animation Relative', the 4 hand IK bones should be set to 'Animation'. I then re-created the virtual bones, there should be 11 in total. Clear the curves on the new skeleton and copy the Curves from 'SK_Als' to the new skeleton. To create the Control Rig I just duplicated the CR_Als and changed the 'Preview Mesh' to the new skeletal mesh.

Here are some screenshots of what needs changing.

![Screenshot (172)](https://github.com/Sixze/ALS-Refactored/assets/132231831/f9ae3a5b-504e-4732-8166-6d73a251f526)
 
![Screenshot (173)](https://github.com/Sixze/ALS-Refactored/assets/132231831/b6c198c5-c440-4380-b714-fa68b344d3d0)

![Screenshot (175)](https://github.com/Sixze/ALS-Refactored/assets/132231831/d1c4caca-9193-447d-bb25-ec0fb6c540e2)
